b2share/modules/records/serializers/schemas/json.py

```diff
@@ -73,9 +75,9 @@ def filter_internal(self, data):
                 g.record_hit['_source'])
 
         if '_deposit' in data['metadata']:
-            if hasattr(g, 'record') and is_deposit(record.model) and current_app.config.get('AUTOMATICALLY_ASSIGN_DOI', False):
+            if hasattr(g, 'record') and is_deposit(record.model):# and current_app.config['AUTOMATICALLY_ASSIGN_DOI']:
                 # add future DOI string
-                data['metadata'].update({'$future_doi': generate_doi(data['metadata']['_deposit']['id']) })
+                data['b2share'] = {'future_doi': generate_doi(data['metadata']['_deposit']['id']) }
```

Comment:
This is to enable the doi to be accessed for the end user before publication.
We should keep this change -HVP